text/0073-event-origin.md

```diff
@@ -0,0 +1,100 @@
+- Start Date: 2023-02-10
+- RFC Type: decision
+- RFC PR: https://github.com/getsentry/rfcs/pull/73
+- RFC Status: draft
+
+# Summary
+
+This RFC aims to give Sentry developers insights into which types of events and transactions our customers use.
+
+# Motivation
+
+We, the SDK developers, would like to get insights into the types of events and transactions
+our customers use, which is only partially possible when writing this.
+
+# Background
+
+While Looker allows queries for `Exception Stack Mechanism Type` to gain insight into
+different error events, it doesn't allow querying for different transaction types. We
+use the SDK integration list to determine which organizations have specific performance
+integrations enabled. The downside is that the SDK sends this list for each event, not
+giving us insights into how many events/transactions/spans stem from a specific parts
+of the SDK. 
+Furthermore, knowing what created an event, transaction, or span helps investigating issues.
+
+
+# Options Considered
+
+For every option, Looker picks up the field, but we don't need to index it and make it searchable in Discover. Amplitude could look at this field as a property when users visit issue or transaction detail pages.
+
+- [Option 1: Event SDK Origin](#option-1)
+- [Option 2: Event Origin](#option-2)
+- [Option 3: Transaction Info Type](#option-3)
+
+
+## Option 1: Event SDK Origin <a name="option-1"></a>
+
+Add a new property to the [SDK interface of the event payload](https://develop.sentry.dev/sdk/event-payloads/sdk/) named `origin` to determine which part of the SDK created the event. 
```

Comment:
I would say that errors can be mostly figured out based on
1) name in SDK interface
2) integration set
3) mechanism
4) contextual tags/message/event type

In addition - does it really matter what created an error? I fear this data will just cause anchoring bias.

I see how this is valuable for spans though - perhaps we add a new attribute to spans to track what generated them (transactions would inherit this). IMO I would like to see this proposal to be transactions/spans only, and focus the options on how we can identify them. This also can help in the UI, showing users what parts of their transaction were auto-instrumented/manually-instrumented. In addition, this difference can help the performance product build new features and performance issues..

Reply:
Thanks for the feedback, @AbhiPrasad. I will update the RFC accordingly.

Reply:
@adinauer, which option would work best for your requirements?

Reply:
I presume we could use `mechanism` more thoroughly for errors, e.g. in logging integrations. I still believe a common solution for errors and transactions would be nice but I don't wanna drag this on just because of that. So we can try the `mechanism` approach first and then revisit if that doesn't work out. Created https://github.com/getsentry/sentry-java/issues/2555